src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs

```diff
@@ -91,19 +92,27 @@ private async Task<IEnumerable<Assembly>> LoadAssembliesInClientAsync(IEnumerabl
                 return loadedAssemblies;
             }
 
-            var assemblies = ((IJSUnmarshalledRuntime)_jsRuntime).InvokeUnmarshalled<object, object, object, object[]>(
-                ReadDynamicAssemblies,
+            var assemblies = ((IJSUnmarshalledRuntime)_jsRuntime).InvokeUnmarshalled<object, object, object, byte[][]>(
+                ReadLazyAssemblies,
                 null,
                 null,
                 null);
 
-            foreach (byte[] assembly in assemblies)
+            var pdbs = ((IJSUnmarshalledRuntime)_jsRuntime).InvokeUnmarshalled<object, object, object, byte[][]>(
+                ReadLazyPDBs,
+                null,
+                null,
+                null);
+
+            foreach (var item in assemblies.Zip(pdbs, (a, p) => new { Assembly = a, PDB = p}))
```

Comment:
An alternative would be to iterate over the indices

Reply:
Yeah, three days ago I thought this pattern was much more reasonable, but present Safia degrees.